corehq/apps/reports/standard/deployments.py

```diff
@@ -543,17 +544,17 @@ def _get_formatted_assigned_location_names(self, user, user_loc_dict):
             loc_name = user_loc_dict.get(loc_id)
             if loc_id == primary_location_id:
                 formatted_loc_names.insert(
-                    0, f'<strong>{loc_name}</strong>'
+                    0, format_html('<strong>{}</strong>', loc_name)
                 )
             else:
-                formatted_loc_names.append(loc_name)
+                formatted_loc_names.append(format_html('{}', loc_name))
 
-        formatted_str = ', '.join(formatted_loc_names[:4])
+        formatted_str = mark_safe(', '.join(formatted_loc_names[:4]))
         html_nodes = [
-            f'<span class="locations-list">{formatted_str}</span>',
+            format_html('<span class="locations-list">{}</span>', formatted_str)
         ]
         if len(formatted_loc_names) > 4:
-            all_str = ', '.join(formatted_loc_names)
+            all_str = mark_safe(', '.join(formatted_loc_names))
```

Comment:
Could `format_html_join` be used here as well?

Reply:
Updated in https://github.com/dimagi/commcare-hq/pull/36286/commits/be8c8b5cdd596df1661f192e41a4c8974d8c03d0 though I had to make the formatted_loc_names a list of list since format_html_join expects a group for arguments for each iteration. I am not sure what the advantage it provides here instead of using mark_safe since we already used format_html for each element in `formatted_loc_names` earlier to have safe string. So, if it sounds reasonable, I'd like to drop the commit that changed to use format_html_join unless there is an advantage to using it.

Reply:
`mark_safe` is easy to use incorrectly because it's easy to accidentally pass in a string that is not safe. `format_html` and `format_html_join` are harder to use incorrectly, and therefore preferable. Best practice is to eliminate all uses of `mark_safe` where the argument is not a literal string that is easily visually evaluated for safety.

Reply:
Thanks for the details @millerdev. I was considering that since I already escaped the location names via format_html earlier, it was now safe to mark the concatenation of those strings as safe.
Seems fine to be double sure.